pkg/game/board.go

```diff
@@ -132,3 +134,140 @@ func (board *board) checkCompleted(
+	var meeples = []elements.MeepleTilePlacement{}
+	var tile = startTile
+	var tileExists bool
+	var score = 0
+	var road feature.Feature
+	var finished bool
+	var err error
+	// check finished on way
+	for roadSide != side.Center && err == nil {
+
+		tile, tileExists = board.GetTileAt(tile.Pos.Add(elements.PositionFromSide(roadSide)))
+		// check if tile exists or loop
+		if !tileExists || tile.Pos == startTile.Pos {
+			// tile does not exist
+			// finish
+			break
+		}
+
+		score++
+		roadSide, err = roadSide.ConnectedOpposite()
+		// check error
+		if err == nil {
+
+			// check for meeple1
+			if tile.Meeple.Side == roadSide {
+				meeples = append(meeples, elements.MeepleTilePlacement{MeeplePlacement: tile.Meeple, PlacedTile: tile})
+			}
+
+			// get road feature
+			road = *tile.GetFeatureAtSide(roadSide) // check isn't needed because it was already checked at legalmoves
+
+			// swap to other end of tile
+			if road.Sides[0] == roadSide {
+				roadSide = road.Sides[1]
+			} else {
+				roadSide = road.Sides[0]
+			}
+
+			// check for meeple2 (other end of road)
+			if tile.Meeple.Side == roadSide {
+				meeples = append(meeples, elements.MeepleTilePlacement{MeeplePlacement: tile.Meeple, PlacedTile: tile})
+			}
+		}
+	}
+
+	finished = (roadSide == side.Center) || (tile.Pos == startTile.Pos)
+	finished = finished && err == nil
+
+	return finished, score, meeples, tile.Pos == startTile.Pos
+}
+
+/*
+Calculates score for road.
+
+returns: ScoreReport
+*/
+func (board *board) ScoreRoadCompletion(tile elements.PlacedTile, road feature.Feature) elements.ScoreReport {
+	var meeples = []elements.MeepleTilePlacement{}
+	var leftSide, rightSide side.Side
+	var score = 1
+	leftSide = road.Sides[0]
+	rightSide = road.Sides[1]
+	var roadFinished = true
+
+	var roadFinishedResult bool
+	var scoreResult int
+	var meeplesResult []elements.MeepleTilePlacement
+	var loopResult bool
+
+	// check meeples on start tile
+	if tile.Meeple.Side == leftSide || tile.Meeple.Side == rightSide {
+		meeples = append(meeples, elements.MeepleTilePlacement{MeeplePlacement: tile.Meeple, PlacedTile: tile})
+	}
+
+	roadFinishedResult, scoreResult, meeplesResult, loopResult = board.CheckRoadInDirection(leftSide, tile)
+	score += scoreResult
+	roadFinished = roadFinished && roadFinishedResult
+	meeples = append(meeples, meeplesResult...)
+
+	if !loopResult {
+		roadFinishedResult, scoreResult, meeplesResult, _ = board.CheckRoadInDirection(rightSide, tile)
+		score += scoreResult
+		roadFinished = roadFinished && roadFinishedResult
+		meeples = append(meeples, meeplesResult...)
+	}
+
+	// -------- start counting -------------
+	if roadFinished {
+		var mostMeeples = 0
+		var scoredPlayers = []uint8{}
+
+		// check who has most meeples naively
+		for _, meepleA := range meeples {
+			var counter = 0
+			for _, meepleB := range meeples {
+				if meepleA.Player.ID() == meepleB.Player.ID() {
+					counter++
```

Comment:
I think it would be more simple to just create a map like {player ID: number of meeples} and after that find all the maximum values. If this works, I don't insist on changing, but you may consider it.

Reply:
changed